Model/CatalogIngestion/Request/Product/DataProcessor.php

```diff
@@ -223,18 +223,19 @@ private function getVariants(int $productId, WebsiteInterface $website): array
      *
      * @param int $productId
      * @param int $storeId
+     * @param int|null $parentProductId
      * @return array
      * @throws LocalizedException
      * @throws FileSystemException
      */
-    private function getBoltProductData(int $productId, int $storeId): array
+    private function getBoltProductData(int $productId, int $storeId, int $parentProductId = null): array
```

Comment:
When we call this function from getRequestProductData
`
$defaultStoreViewBoltProductData = $this->getBoltProductData($productId, (int)$defaultStore->getId());`

it still possible that productId is a child product. Can we cover this case and include test?